### ISSUE TITLE
southpark-example: getting empty result

### ISSUE THREAD

**Issue by Kavan72** [jina-ai/examples#160]
i've a dataset something like `jsonl`

jina version: 0.4.10 (master branch)
python version:  3.7.8

My changes:

`encode.yml`: -
  - removed `max_length: 96`

`text_loader.py`: -
```
class TextExtractor(BaseCrafter):
    def craft(self, text: str, *args, **kwargs) -> Dict:
        data_dict = json.loads(text)
        return dict(weight=1., text=data_dict['file_data'], meta_info=data_dict['file_id'].encode('utf8'))
```
`app.py`:-

```
def print_topk(resp, sentence):
    for d in resp.search.docs:

        for idx, match in enumerate(d.matches):
            score = match.score.value
            if score < 0.0:
                continue
            file_id = match.meta_info.decode()
            print('idx={} score={} file_id={}'.format(idx, score, file_id))
```

here is full log: https://pastebin.com/NgKNJ4X4

**Comment by YueLiu-jina** [comment]
Hi @Kavan72 ,

The current Southpark code is based on Jina version 0.4.1, so the problem is likely from Jina's version. Please use `pip install --upgrade -r requirements.txt` or `pip install jina==0.4.1` to upgrade it.

**Comment by nan-wang** [comment]
hi, @Kavan72 . Please note the version requirements for each examples. We are refactoring codes quickly with the master branch and therefore we lock the version for the examples.

**Comment by Kavan72** [comment]
@nan-wang @YueLiu-jina thanks for your response
reason is using master branch is, there was a some problem in request's generator. and it was fixed on 0.4.5, but it's okay
My another question is why i'm getting only 50 results even if i pass `top-k = 100` on jina 0.4.1.

- i want to search some words or sentence on files. so, i did this steps
- i did index on 100 files like this `python app.py -t index -n 100`
- i run jina for query like this `python app.py -t query -n 100 -k 100`

from this, i'm getting only 50 results inside of 100

**Comment by JoanFM** [comment]
Hey @Kavan72 

**Things to consider about top_k**
It is quite an strange behavior, but it could happen that the system does not return **top_k** results.
The system is returning **top_k** chunks per each query chunk . Then they are aggregated for all the chunks in the document, but it could happen that some chunks matches 10 chunks but not necessarily from 10 different documents (so you may match 10 chunks from the same document). Then, when the **top_k** documents want to be extracted, it could happen that some **top_k** results bucket is missing.

I hope it clarifies

**Comment by Kavan72** [comment]
hey @JoanFM thanks for your response

I got it, what you are trying to say. i'm trying to build semantic search system. Something like, search words or sentence on each docs and get the result of each docs as per given encoder.

**Comment by Kavan72** [comment]
fixed on 0.5.5. thank you
